runtime/onert/core/CMakeLists.txt

```diff
@@ -13,6 +13,9 @@ target_link_libraries(onert_core PRIVATE nnfw_coverage)
 target_link_libraries(onert_core PRIVATE dl ${LIB_PTHREAD})
 target_link_libraries(onert_core PRIVATE jsoncpp)
 target_link_libraries(onert_core INTERFACE ruy_instrumentation)
+if (ANDROID)
+  target_link_libraries(onert_core INTERFACE log)
```

Comment:
Is it necessary?

Reply:
@YongseopKim Android build fails without this line. I'm not sure this is the correct way to fix linking error, but it works. 

**Android Linking Dependency** 

onert_core -> cker -> ruy -> cpuinfo -> clog -> log

onert_core needs to link `liblog.so`, but it fails without above line.

Reply:
I got it, thank you for your answer.

Reply:
Then can't we link `log` to ruy`?

Reply:
@hseok-oh Same error is generated when I linked `log` to `ruy`. 

```bash
/ndk/toolchains/llvm/prebuilt/linux-x86_64/lib/gcc/aarch64-linux-android/4.9.x/../../../../aarch64-linux-android/bin/ld: warning: liblog.so, needed by libonert_core.so, not found (try using -rpath or -rpath-link)
```

Reply:
That's odd, though...

Reply:
Then maybe cpuinfo's header file requires `log`, and onert_core uses cpuinfo's header. Did you link `log` to `ruy` with `PRIVATE` keyword?

Reply:
@hseok-oh I've tried both `INTERFACE` and `PUBLIC` keywords but the error remains.

Reply:
@hseok-oh @wateret Adding `-fuse-ld=gold` to linker flags solves this issue. I found this solution from https://github.com/android/ndk/issues/1088. I'll update this PR soon.

Reply:
@periannath If that has too much work and it is sure that is a bug, we may leave some comments with the current changes.

Reply:
@wateret It seems that this is caused by bug in BFD linker. To solve this problem, we have to use gold linker instead of BFD linker in android build. The code change is small but I'm not sure about its side-effect. 
If it is okay, I want to review this PR as is and discuss about android linker in another issue or PR.

Reply:
@periannath Sure, but could you leave some comments here?

Reply:
@wateret I've added comments as you suggested.